File: msgraph-mail/ruby/graphrubyv4/models/microsoft/graph/inference_classification.rb
require '../../models'
require '../microsoft'
require './graph'
require 'microsoft_kiota_abstractions'
require_relative './graphrubyv4::_models::_microsoft::_graph::_entity'

module Graphrubyv4::Models::Microsoft::Graph
    class InferenceClassification < Graphrubyv4::Models::Microsoft::Graph::Entity
        include MicrosoftKiotaAbstractions::Parsable
        ## 
        # A set of overrides for a user to always classify messages from specific senders in certain ways: focused, or other. Read-only. Nullable.
        @overrides
        ## 
        ## Gets the overrides property value. A set of overrides for a user to always classify messages from specific senders in certain ways: focused, or other. Read-only. Nullable.
        ## @return a inference_classification_override
        ## 
        def  overrides
            return @overrides
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return super.merge({
                "overrides" => lambda {|o, n| o.overrides = n.get_collection_of_object_values(Graphrubyv4::Models::Microsoft::Graph::InferenceClassificationOverride) },
            })
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            super
            writer.write_collection_of_object_values("overrides", @overrides)
        end
        ## 
        ## Sets the overrides property value. A set of overrides for a user to always classify messages from specific senders in certain ways: focused, or other. Read-only. Nullable.
        ## @param value Value to set for the overrides property.
        ## @return a void
        ## 
        def  overrides=(overrides)
            @overrides = overrides
        end
    end
end

File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/models/microsoft/graph/InferenceClassification.java
package graphjavav4.utilities.models.microsoft.graph;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
public class InferenceClassification extends Entity implements Parsable {
    /** A set of overrides for a user to always classify messages from specific senders in certain ways: focused, or other. Read-only. Nullable.  */
    private java.util.List<InferenceClassificationOverride> _overrides;
    /**
     * Instantiates a new inferenceClassification and sets the default values.
     * @return a void
     */
    public InferenceClassification() {
        super();
    }
    @javax.annotation.Nonnull
    public static InferenceClassification createFromDiscriminatorValue(@javax.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new InferenceClassification();
    }
    /**
     * Gets the overrides property value. A set of overrides for a user to always classify messages from specific senders in certain ways: focused, or other. Read-only. Nullable.
     * @return a inferenceClassificationOverride
     */
    @javax.annotation.Nullable
    public java.util.List<InferenceClassificationOverride> getOverrides() {
        return this._overrides;
    }
    /**
     * The deserialization information for the current model
     * @return a Map<String, BiConsumer<T, ParseNode>>
     */
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getFieldDeserializers() {
        return new HashMap<>(super.getFieldDeserializers()) {{
            this.put("overrides", (o, n) -> { ((InferenceClassification)o).setOverrides(n.getCollectionOfObjectValues(InferenceClassificationOverride::createFromDiscriminatorValue)); });
        }};
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeCollectionOfObjectValues("overrides", this.getOverrides());
    }
    /**
     * Sets the overrides property value. A set of overrides for a user to always classify messages from specific senders in certain ways: focused, or other. Read-only. Nullable.
     * @param value Value to set for the overrides property.
     * @return a void
     */
    public void setOverrides(@javax.annotation.Nullable final java.util.List<InferenceClassificationOverride> value) {
        this._overrides = value;
    }
}
